ProcessMaker/processmaker
# Integrate Gmail API Email Server

## Issue & Reproduction Steps

Ticket: [FOUR-6104](https://processmaker.atlassian.net/browse/FOUR-6104)

As a System Administrator, I want to enable an email server using the Gmail API so that my platform utilizes my preferred and familiar email service

This PR updates the Settings UI based on the [Email Server PRD](https://processmaker.atlassian.net/wiki/spaces/PM4/pages/2761064449/Email+Server+Options+-+Office365+Modern+Auth+and+Gmail+API) specifications as well as adds a Custom Gmail Driver for Gmail Api support.

## Solution

- Remove Pagination
- Update the placement and verbiage of tab buttons
- Update 'Action' icons
- Adds the 'Gmail' option in the UI for the 'Mailer Driver' Model
- Implements Custom Gmail Driver
- Adds migration 

## UI Updates

**Before**

![Screen Shot 2022-07-14 at 11 07 28 AM](https://user-images.githubusercontent.com/52755494/179053935-eff9820a-472f-45e7-b4a6-a206f9d77759.png)
![Screen Shot 2022-07-14 at 11 07 39 AM](https://user-images.githubusercontent.com/52755494/179053975-d779a316-99ea-4af3-b0ce-61a02b326c9d.png)

**After**
![Screen Shot 2022-07-14 at 11 15 45 AM](https://user-images.githubusercontent.com/52755494/179054016-ab28fa26-4512-431d-b51c-fd18dd1b35f6.png)

![Screen Shot 2022-07-14 at 11 15 52 AM](https://user-images.githubusercontent.com/52755494/179054038-9209974f-036e-488e-8556-69a9f5588300.png)

## How to Test

**Testing UI**
1. Review the [PRD](https://processmaker.atlassian.net/wiki/spaces/PM4/pages/2761064449/Email+Server+Options+-+Office365+Modern+Auth+and+Gmail+API)
1. Ensure the following packages are installed - 
    - Auth Package
    - Send Email Connector [PR](https://github.com/ProcessMaker/connector-send-email/pull/209) `feature/FOUR-6104`
    - Actions By Email
    - Docusign Connector
2. Run `php artisan migrate` 
3. Within the settings tabs ensure UI displays properly
   - Buttons are displayed properly
   - Add / Remove Email Servers works as expected
   - Configurations Update as expected. etc...

## Related Tickets & Packages
- Ticket: [FOUR-6104](https://processmaker.atlassian.net/browse/FOUR-6104)
- [Email Server PRD](https://processmaker.atlassian.net/wiki/spaces/PM4/pages/2761064449/Email+Server+Options+-+Office365+Modern+Auth+and+Gmail+API)
- [Send Email PR](https://github.com/ProcessMaker/connector-send-email/pull/209) `feature/FOUR-6104`

## Code Review Checklist
